DevelopingSpace/starchart
# Implement domain creation page UI

Resolves #48 

- Implemented `domain/dns record` creation form UI based on [figma](https://www.figma.com/file/XAcnMIC1y4qumWinSalHI9/Starchart-UI?node-id=2175%3A3583&t=oM3dYngtGD0QSrZf-0) design
- Added a simple action to create a `domain/dns record` and redirect to domain edit page. This is simply for testing purposes and would be replaced in the future
- Added error checking for `Type` field. If any value other than the ones allowed is sent, it will show an error. (Note that to do this, you would need to manually edit the html) Any other validations can be added following this pattern

> Domain should be replaced with DNS Record and will be fully handled in #245 

<img width="960" alt="image" src="https://user-images.githubusercontent.com/67077705/221165674-59376cb3-1ede-4194-84bf-080cbf2b6017.png">

### Steps to test
- Run app locally
- Run `npm run db:studio` at the same time to see the `DNS Records/Domains` being created
- Navigate to `/domains/new`
- Check form style
- Check tooltips text
- Ensure `Domain name`, `Type`, and `Value` fields are required
- Try entering different values and check that `DNS Records/Domains` are created for valid data
